src/plugins/xerces/deserializer.cpp

```diff
@@ -0,0 +1,141 @@
+/**
+ * @file
+ *
+ * @brief deseialization implementation for xerces plugin
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#include "deserializer.hpp"
+
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <memory>
+
+#include <xercesc/dom/DOM.hpp>
+#include <xercesc/dom/DOMAttr.hpp>
+#include <xercesc/dom/DOMDocument.hpp>
+#include <xercesc/dom/DOMError.hpp>
+#include <xercesc/dom/DOMException.hpp>
+#include <xercesc/dom/DOMImplementation.hpp>
+#include <xercesc/dom/DOMNamedNodeMap.hpp>
+#include <xercesc/dom/DOMNode.hpp>
+#include <xercesc/parsers/XercesDOMParser.hpp>
+
+#include <key.hpp>
+
+XERCES_CPP_NAMESPACE_USE
+using namespace std;
+using namespace kdb;
+
+/*
+ + Smart pointer helper functions to ease the pain of dealing between XMLCh and std::string
+ */
+
+// no need to expand this, just keep it in one line
+// TODO move to external file if also used for serializer
+// clang-format off
+struct XmlChDeleter { void operator() (XMLCh * ptr) { XMLString::release (&ptr); } };
```

Comment:
Please have a look at the RAII pattern, it looks like you overuse smart pointers.